src/main/java/com/navercorp/openwhisk/intellij/common/whisk/model/Runtime.java

```diff
@@ -16,51 +16,60 @@
 
 package com.navercorp.openwhisk.intellij.common.whisk.model;
 
+/**
+ *  Set the type of runtime based on upstream
+ *    - https://github.com/apache/openwhisk/blob/master/ansible/files/runtimes.json
+ *
+ *  The runtime used for each endpoint may be different, so use it according to the policy of the endpoint users are using.
+ */
 public enum Runtime {
-    @Deprecated NODE_6("nodejs", ".js", "6"),
-    @Deprecated NODE_8("nodejs", ".js", "8"),
-    NODE_10("nodejs", ".js", "10"),
-    NODE_12("nodejs", ".js", "12"),
-    PYTHON_2("python", ".py", "2"),
-    PYTHON_3("python", ".py", "3"),
-    JAVA("java", ".java", "default"),
-    SWIFT_3_1_1("swift", ".swift", "3.1.1"),
-    SWIFT_4_2("swift", ".swift", "4.2"),
-    PHP_7_2("php", ".php", "7.3"),
-    GO_1_11("go", ".go", "1.11"),
-    RUBY_2_5("ruby", ".rb", "2.5"),
-    SEQUENCE("sequence", "", ""),
-    DOCKER("blackbox", "", "");
+    NODE_6("nodejs", ".js", "6", false, true),
+    NODE_8("nodejs", ".js", "8", false, true),
+    NODE_10("nodejs", ".js", "10", true, false),
+    NODE_12("nodejs", ".js", "12", false, false),
+    NODE_14("nodejs", ".js", "14", false, false),
+    PYTHON_2("python", ".py", "2", false, false),
+    PYTHON_3("python", ".py", "3", true, false),
+    JAVA("java", ".java", "8", true, false),
+    SWIFT_3_1_1("swift", ".swift", "3.1.1", false, true),
+    SWIFT_4_2("swift", ".swift", "4.2", true, false),
+    SWIFT_5_1("swift", ".swift", "5.1", false, false),
+    PHP_7_3("php", ".php", "7.3", false, false),
+    PHP_7_4("php", ".php", "7.4", true, false),
+    GO_1_11("go", ".go", "1.11", true, false),
+    RUBY_2_5("ruby", ".rb", "2.5", true, false),
+    SEQUENCE("sequence", "", "", true, false),
+    DOCKER("blackbox", "", "", true, false);
 
     private String name;
     private String extension;
     private String version;
+    private boolean defaultRuntime;
+    private boolean deprecated;
 
-    Runtime(String name, String extension, String version) {
+    Runtime(String name, String extension, String version, boolean defaultRuntime, boolean deprecated) {
```

Comment:
@style95 

Yes, so node6 and node8's deprecated value is true

Reply:
ok at some point, it would be great to just remove them as upstream no longer includes them.